runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineWorkerPoolOptions.java

```diff
@@ -112,30 +108,30 @@ public String getAlgorithm() {
 
   void setDiskSizeGb(int value);
 
-  /**
-   * Docker container image that executes Dataflow worker harness, residing in Google Container
-   * Registry.
-   */
-  @Default.InstanceFactory(WorkerHarnessContainerImageFactory.class)
+  /** Container image used as Dataflow worker harness image. */
+  /** @deprecated Use {@link #getSdkContainerImage} instead. */
   @Description(
-      "Docker container image that executes Dataflow worker harness, residing in Google "
-          + " Container Registry.")
+      "Container image used to configure a Dataflow worker. "
+          + "Can only be used for official Dataflow container images.")
+  @Deprecated
   @Hidden
   String getWorkerHarnessContainerImage();
 
+  /** @deprecated Use {@link #setSdkContainerImage} instead. */
+  @Deprecated
+  @Hidden
   void setWorkerHarnessContainerImage(String value);
 
   /**
-   * Returns the default Docker container image that executes Dataflow worker harness, residing in
-   * Google Container Registry.
+   * Container image used to configure SDK execution environment on worker, refered to as the SDK
+   * harness image.
    */
-  class WorkerHarnessContainerImageFactory implements DefaultValueFactory<String> {
```

Comment:
Now, the value of either image will not have a default. I think this is an acceptable "breaking" change since the returned URL is not usable as-is.